server_transport/src/wstcp.c

```diff
@@ -3369,6 +3370,49 @@ xUNUSED static const char * zz_encrypt(const char * data, char * buf, int len) {
     ism_common_toBase64((char *)encdata, (char *)b64data, blklen+1);
     ism_common_strlcpy(buf, (const char *)b64data, len);
     return buf;
+
+#else
+
+    EVP_CIPHER_CTX *ctx;
+    int len;
+    int ciphertext_len;
+
+    /* Create and initialise the context */
+    if(!(ctx = EVP_CIPHER_CTX_new()))
+        TRACE(7, "EVP_CIPHER_CTX_new failed\n");
```

Comment:
Trace level is set to 5 by default i.e. any trace messages with number 6 or higher aren't recorded anywhere). It's a rare case but if these openssl functions fail, no ret code is returned (which would be a bigger change) and bad/no data is written to the buffer - if anyone was to try and debug this having at least some messages in the trace would help.

At a minimum lower the trace level, but how big a change would it be to make the function fail if the evp functions fail and the caller do something sensible with the error? If there is nothing sensible the caller can do consider creating an FFDC (a bunch of diagnostic data) and whether to make that FFDC fatal (i.e. stop the server) or not. Example ffdc:
https://github.com/eclipse/amlen/blob/main/server_admin/src/authentication.c#L1082

Reply:
I see, lowering trace level below or equal to 5 is useful in that case.

FFDC won't require here as we do not want to do any sensible with the error as if encrypt/decrypt fails then caller function must handle that failure, not the EV failure requires to do anything else than logging it.

As a part of this comment, I will change trace level to 5. Please suggest your thought.

Reply:
At the moment - the caller doesn't know - there's no return code to say the encryption failed.

Reply:
I see none of the caller working on failure of encrypt or decrypt function.
Do you think, aborting server on any EVP API call failure is good idea here ?

Reply:
returning NULL on each EVP function failure.